prepare_data.py

```diff
@@ -0,0 +1,82 @@
+import json
+import os
+from datetime import datetime
+
+import requests
+import xlsxwriter
+
+
+def fetch_data():
+    url = "https://ckan2.multimediagdansk.pl/gpsPositions?v=2"
+    response = requests.get(url)
+    if response.status_code == 200:
+        data = response.json()
+        parse_data(data_dir, data)
+    else:
+        print("Failed to retrieve data")
+
+
+def parse_data(data_dir: str, data: dict):
+    data_datetime = None
+    try:
+        data_datetime = datetime.strptime(data["lastUpdate"], "%Y-%m-%dT%H:%M:%SZ")
+    except:
+        print("Invalid JSON")
+    # reise
+    file_name = data_datetime.strftime("%Y-%m-%d") + ".json"
+    create_file_if_not_exist(data_dir, file_name)
+
+    file_path = os.path.join(data_dir, file_name)
+    with open(file_path, "r+") as f:
+        f_data = json.load(f)
+        current_time = data_datetime.strftime("%H:%M:%S")
+        if current_time in f_data.keys():
+            print("This update exist")
+        else:
+            vehicles = []
+            for v in data["vehicles"]:
+                scheduledTripStartTime = None
+                should_append = True
+                if v["scheduledTripStartTime"]:
```

Comment:
oh noo, use something like: 
```suggestion
                if not v["scheduledTripStartTime"]:
                    return 
```